### PR TITLE
Convert PhaseChange to new factory

### PR DESCRIPTION
**Pull request by wthrowe**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Executables/Burgers/EvolveBurgers.hpp [hunk 1]
```diff
@@ -271,44 +296,12 @@ struct EvolutionMetavars {
...
-  using phase_change_tags_and_combines_list =
-      PhaseControl::get_phase_change_tags<phase_changes>;
-
   using const_global_cache_tags = tmpl::list<
       initial_data_tag, Tags::EventsAndTriggers,
       tmpl::conditional_t<use_dg_subcell,
                           tmpl::list<Burgers::fd::Tags::Reconstructor>,
                           tmpl::list<>>,
-      PhaseControl::Tags::PhaseChangeAndTriggers<phase_changes>>;
+      PhaseControl::Tags::PhaseChangeAndTriggers>;
```
Comment (nilsvu) [inline src/Evolution/Executables/Burgers/EvolveBurgers.hpp]: We collect cache tags from actions, right? In that case I think you can list this tag in the ExecutePhaseChange action and delete it here. Same for the EventsAndTriggers actually I think.

**Review by nilsvu — COMMENTED** [review]
This is great. I find simplifying metavars is extremely valuable.

File: src/Evolution/Executables/Burgers/EvolveBurgers.hpp (hunk 1 above)
Comment (wthrowe) [reply src/Evolution/Executables/Burgers/EvolveBurgers.hpp]: Good idea.  Done both.

**Review by kidder — APPROVED** [review]
This severely conflicts with #3453  which should probably therefore be closed; is there anything we want to salvage from that PR?

**Review by nilsvu — APPROVED** [review]
Merging now to avoid rebase conflicts with keeping this open for long.

@kidder yes I think Jordan added lots of useful docs in that PR, and we should document/refactor the `registration_list` metavar.

**Comment by wthrowe** [comment]
Yeah, the registration-related stuff from there is still useful.

**Pull request merged**